### DISCUSSION TITLE
building a project that uses the pde tools (annotations) is that really only supported from Tycho 5.0 on?

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-tycho/tycho#3780] [General]
We have a jface patch that uses the pde tools so i get this:

workspace/org.eclipse.jface/src/org/eclipse/jface/viewers/CheckboxTableViewer.java:[23] 
[ERROR] 	import org.eclipse.pde.api.tools.annotations.NoExtend;
[ERROR] 	       ^^^^^^^^^^^^^^^
[ERROR] The import org.eclipse.pde cannot be resolved

It works inside in eclipse because of the container that is added:

https://github.com/Servoy/servoy-eclipse/blob/master/org.eclipse.jface/.classpath#L4

now i do read in the release notes of 5.0 that this is a plugin that we can add for 5.0?

https://github.com/eclipse-tycho/tycho/blob/main/RELEASE_NOTES.md

so with 4.0 its not possible yet to build something like this with tycho?

**Comment by laeubi** [comment]
Well first it often don't harm to simply try it out as many features (if not all) are backported to Tycho 4 at the moment, this might change when something prevents us to do so (e.g major refactoring happens, new maven/java versions used ...).

Now if you look [here](https://mvnrepository.com/artifact/org.eclipse.tycho/tycho-apitools-plugin/4.0.7) you will see the plugin is actually there and eclipse platform is also using this so in general I don't see a reason for this not working in Tycho 4 line.

**Comment by jcompagner** [comment]
ahh thx, yeah i was looking in the release notes and there it seems to be that it was under the tycho 5 section ..

But i will try this out, thx.
